### PR TITLE
Per Document Config

### PR DESCRIPTION
**Pull request by loofou**
This is currently incomplete.

This change introduces a new SWIM menu for each actor and item. This menu can be used to set type-specific configuration and store them automatically in flags.
It also introduces a way to load and store ActiveEffects via this config menu. Those effects can then be re-used later.

Right now it's not fully tested and probably has a lot of quirks. Merging is not recommended, but if we want to use the new menu already, we should probably disable the "gear" config for now (just comment out the current options related to ammo).

Actor menu is disabled but should technically work the same as items and could be activated relatively easy.

### CODE REVIEW

File: swim/scripts/init.js [hunk 1]
```diff
@@ -154,6 +155,27 @@ Hooks.on(`ready`, () => {
     warpgate.event.watch("SWIM.updateCombat-previousTurn", gm_relay.combat_previousTurn, swim.is_first_gm)
     warpgate.event.watch("SWIM.updateCombat-nextTurn", gm_relay.combat_nextTurn, swim.is_first_gm)
     warpgate.event.watch("SWIM.updateCombat-currentTurn", gm_relay.combat_currentTurn, swim.is_first_gm)
+
+    //SWIM per-actor/item config header button
+    if (game.user.isGM) {
```
Comment (SalieriC) [inline swim/scripts/init.js:160]:
```js
if (game.user.isGM || game.settings.get('swim', 'allowUserConfig') {
```
I will add the setting for that so GMs can decide whether or not they want to allow non-GMs to edit the config.

Comment (loofou) [reply swim/scripts/init.js:160]: You want me to do it or you do it in this branch?

**Review by SalieriC — APPROVED** [review]
Looking grand so far. =)

**Comment by loofou** [comment]
This should now be ready to merge into v10. Give it one more quick review, please. If you don't see anything problematic, we can go ahead and merge this in. I can then start to look into how we want to improve the Ammo system :)

**Pull request merged**